src/securejoin/mod.rs

```diff
@@ -262,22 +293,24 @@ async fn securejoin(context: &Context, qr: &str) -> Result<ChatId, JoinError> {
     let qr_scan = check_qr(context, &qr).await;
     let invite = QrInvite::try_from(qr_scan)?;
 
-    context.bob.start_protocol(context, invite.clone()).await?;
-
-    match invite {
-        QrInvite::Contact { .. } => {
+    match context.bob.start_protocol(context, invite.clone()).await? {
+        StartedProtocolVariant::SetupContact => {
             // for a one-to-one-chat, the chat is already known, return the chat-id,
             // the verification runs in background
             let chat_id = chat::create_by_contact_id(context, invite.contact_id())
                 .await
                 .map_err(JoinError::UnknownContact)?;
             Ok(chat_id)
         }
-        QrInvite::Group { ref grpid, .. } => {
-            // for a group-join, wait until the secure-join is done and the group is created
-            while !context.shall_stop_ongoing().await {
-                async_std::task::sleep(Duration::from_millis(50)).await;
-            }
+        StartedProtocolVariant::SecureJoin {
+            ongoing_receiver,
+            group_id,
+        } => {
+            // for a group-join, wait until the protocol is finished and the group is created
+            ongoing_receiver
+                .recv()
+                .await
+                .map_err(|_| JoinError::OngoingSenderDropped)?;
```

Comment:
I was also tempted to just `.ok()` this result, but I think this is better.  Though yet another error variant is a bit annoying, but hey.